packages/fiori/src/SideNavigation.js

```diff
@@ -0,0 +1,239 @@
+import SideNavigationTemplate from "./generated/templates/SideNavigationTemplate.lit.js";
+import SideNavigationItemPopoverContentTemplate from "./generated/templates/SideNavigationItemPopoverContentTemplate.lit.js";
+
+// Styles
+import SideNavigationCss from "./generated/themes/SideNavigation.css.js";
+
+/**
+ * @public
+ */
+const metadata = {
+	tag: "ui5-side-navigation",
+	managedSlots: true,
+	properties: /** @lends sap.ui.webcomponents.fiori.SideNavigation.prototype */ {
+		/**
+		 * Defines whether the <code>ui5-side-navigation</code> is expanded or collapsed.
+		 *
+		 * @public
+		 * @type {boolean}
+		 * @defaultvalue false
+		 */
+		collapsed: {
+			type: Boolean,
+		},
+
+		/**
+		 * @private
+		 */
+		_popoverContent: {
+			type: Object,
+		},
+	},
+	slots: /** @lends sap.ui.webcomponents.fiori.SideNavigation.prototype */ {
+		/**
+		 * Defines the items in the <code>ui5-side-navigation</code>.
+		 *
+		 * @public
+		 * @slot
+		 */
+		"default": {
+			propertyName: "items",
+			invalidateParent: true,
+			type: HTMLElement,
+		},
+
+		/**
+		 * Defines the fixed items in the bottom of the <code>ui5-side-navigation</code>.
+		 *
+		 * @public
+		 * @slot
+		 */
+		fixedItems: {
+			type: HTMLElement,
+			invalidateParent: true,
+		},
+	},
+	events: /** @lends sap.ui.webcomponents.fiori.SideNavigation.prototype */ {
+		/**
+		 * Fired when the selection has changed via user interaction
+		 *
+		 * @event sap.ui.webcomponents.fiori.SideNavigation#selection-change
+		 * @param {HTMLElement} item the clicked item.
+		 * @public
+		 */
+		"selection-change": {
+			item: {
+				type: HTMLElement,
+			},
+		},
+	},
+};
+
+/**
+ * @class
+ *
+ * <h3 class="comment-api-title">Overview</h3>
+ *
+ *
+ * <h3>Usage</h3>
+ *
+ * <code>ui5-side-navigation</code> is used as a standard menu in applications. In order to add menu items use <code>ui5-side-navigation-items</code>.
+ *
+ * For the <code>ui5-side-navigation</code>
+ * <h3>ES6 Module Import</h3>
+ *
+ * <code>import @ui5/webcomponents/dist/SideNavigation.js";</code>
+ *
+ * @constructor
+ * @author SAP SE
+ * @alias sap.ui.webcomponents.fiori.SideNavigation
+ * @extends UI5Element
+ * @tagname ui5-side-navigation
+ * @since 1.0.0-rc.8
+ * @appenddocs SideNavigationItem
+ * @public
+ */
+class SideNavigation extends UI5Element {
+	static get metadata() {
+		return metadata;
+	}
+
+	static get render() {
+		return litRender;
+	}
+
+	static get styles() {
+		return SideNavigationCss;
+	}
+
+	static get template() {
+		return SideNavigationTemplate;
+	}
+
+	static get staticAreaTemplate() {
+		return SideNavigationItemPopoverContentTemplate;
+	}
+
+	static async onDefine() {
+		await Promise.all([
+			Tree.define(),
+			ResponsivePopover.define(),
+		]);
+	}
+
+	onBeforeRendering() {
+		if (this.collapsed) {
+			this._walk(current => {
+				if (current.items) {
+					current.items.forEach(currentSubItem => {
+						if (currentSubItem.selected) {
+							current.selected = true;
```

Comment:
This changes the public API from inside, without user interaction. Only one item should have the "selected" attribute at a time. You need to only set the "selected" attribute of the tree items and list items that are used internally. Probably a good solution would be something like `_buildPopoverContent` where you build items with the item and selected properties